parser.js

```diff
@@ -247,7 +247,18 @@ const getComponentExternalDb = (
         ) {
           continue;
         }
-        const dbName = dbIdentifiersDict[fcomponent]["dbname_map"][header];
+        const crossReferencesArray = Object.values(crossReferencesDict);
+        const referenceData = crossReferencesArray.find((item) =>
+          item.headers.includes(header)
+        );
+        const dbName = referenceData ? referenceData.name : "";
```

Comment:
I haven't tested this yet, so just by looking I would think:
```suggestion
        const dbName = referenceData ? referenceData.db : "";
```
or perhaps even nicer to rename `.db` to something nicer such as `.dbName`